### PR TITLE
Uncolored clouds and PointNet baseline

### PR DESCRIPTION
**Pull request by Vynikal**
### Uncolored clouds
The motivation is to use the model as an augmentation of FLAIR dataset without using inaccurate cloud colorizing. As the attributes are different, it's necessary to write a new dataset description and pre-transform. Also rgb normalizing is now conditional to to rgb attribute. NoRGB separate experiments were created.

### PointNet baseline
Taking advantage of the modular design of Myria3D, just another model to compare with, nothing more, nothing less. Also coming with the NoRGB variant.

The two NoRGB models were trained and tested, more info in the forked repository readme, also including trained checkpoints.

It might not be perfect, feel free to point out the issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,37 +1,60 @@
 <div align="center">
 
 # Myria3D: Aerial Lidar HD Semantic Segmentation with Deep Learning
+# Fork adapted to train/infer on non-colorized data
```
Comment (leavauchier) [inline README.md:4]: Can you remove the parts that are relate to your fork from this MR please, as it won't be relevant once merged

File: README.md [hunk 2]
```diff
@@ -1,37 +1,60 @@
...
 
-[![](https://shields.io/badge/-Lightning--Hydra--Template-017F2F?style=flat&logo=github&labelColor=303030)](https://github.com/ashleve/lightning-hydra-template)
+The training was performed on a laptop with 3070Ti GPU (8GB VRAM), 32 GB RAM and i7-12700H. Batch sizes were adapted to the specifications.
+PointNet implementation subsamples the 50x50m tiles to 4096 points, then upsamples with 1-nn. RandLaNet remains unchanged.
 
-[![CICD](https://github.com/IGNF/myria3d/actions/workflows/cicd.yaml/badge.svg)](https://github.com/IGNF/myria3d/actions/workflows/cicd.yaml)
-[![Documentation Build](https://github.com/IGNF/myria3d/actions/workflows/gh-pages.yml/badge.svg)](https://github.com/IGNF/myria3d/actions/workflows/gh-pages.yml)
-</div>
-<br><br>
+To train PointNet:
```
Comment (leavauchier) [inline README.md:14]: Infos on how to train are in the documentation, can you move your update to this part? eg. [here](https://github.com/IGNF/myria3d/blob/main/docs/source/guides/train_new_model.md)

Same for the inference, see [here](https://github.com/IGNF/myria3d/blob/main/docs/source/tutorials/make_predictions.md)

File: README.md [hunk 3]
```diff
@@ -1,37 +1,60 @@
...
-> &rarr; A stable, production-ready version of Myria3D is tracked by a [Production Release](https://github.com/IGNF/myria3d/releases/tag/prod-release-tag). In the release's assets are a trained multiclass segmentation model as well as the necessary configuration file to perform inference on French "Lidar HD" data. Those assets are provided for convenience, and are subject to change in time to reflect latest model training.
+(to achieve better results add `predict.subtile_overlap=25`)
+
+
+To infer RandLaNet model:
+
+`python run.py task.task_name='predict' predict.src_las='/path/to/las' predict.output_dir='/path/to/output' datamodule.epsg=2154 predict.ckpt_path='${hydra:runtime.cwd}/trained_model_assets/pointnet_norgb_epoch_020.ckpt' trainer.accelerator=gpu predict.gpus=[0]`
 ___
 
+# Comparisons
```
Comment (leavauchier) [inline README.md:36]: I think that your comparisons can have a place in the documentation too, provided that you update the links so that the images are stored in our repo as well (in case yours is modified)

File: myria3d/models/modules/pointnet.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
+import torch
```
Comment (leavauchier) [inline myria3d/models/modules/pointnet.py:1]: This needs testing, tests can be added in https://github.com/IGNF/myria3d/blob/main/tests/myria3d/models/modules/ (see https://github.com/IGNF/myria3d/blob/main/tests/myria3d/models/modules/test_randla_nets.py for an example)

File: myria3d/pctl/points_pre_transform/lidar_hd_norgb.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# function to turn points loaded via pdal into a pyg Data object, with additional channels
```
Comment (leavauchier) [inline myria3d/pctl/points_pre_transform/lidar_hd_norgb.py:1]: this needs tests as well, that can be added in https://github.com/IGNF/myria3d/tree/main/tests/myria3d/pctl/transforms

**Review by leavauchier — CHANGES_REQUESTED** [review]
Hi @Vynikal, thanks a lot for your contribution!
It looks mostly ok to me, and I think it would be great to have this added to myria3d!

Before that, it needs some changes related to your readme and missing tests

**Comment by Vynikal** [comment]
Hi, as this was done during an internship and I've since shifted my work elsewhere, I won't be able to provide further changes. Feel free to adapt the suggestions in any way you see fit.
